RecoHGCal/TICL/plugins/HGCGraph.cc

```diff
@@ -0,0 +1,97 @@
+// Author: Felice Pantaleo - felice.pantaleo@cern.ch
+// Date: 11/2018
+// Copyright CERN
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "RecoHGCal/TICL/interface/TICLConstants.h"
+
+#include "HGCDoublet.h"
+#include "HGCGraph.h"
+
+void HGCGraph::makeAndConnectDoublets(const std::vector<std::vector<std::vector<unsigned int>>> &h,
+                                      int nEtaBins, int nPhiBins,
+                                      const std::vector<reco::CaloCluster> &layerClusters,
+                                      int deltaIEta, int deltaIPhi, float minCosTheta,
+                                      float minCosPointing, int missing_layers) {
+  isOuterClusterOfDoublets_.clear();
```

Comment:
should check memory performance - might want to use `swap()` or `shrink_to_fit()` to avoid hoarding memory after high-occupancy events